aperturedb/Connector.py

```diff
@@ -234,9 +234,10 @@ def _connect(self):
 
         self.connected = True
 
-    def _query(self, query, blob_array = []):
+    def _query(self, query, blob_array = [], check_status=True):
 
-        self._check_session_status()
```

Comment:
Sure. I noticed `_authenticate` also uses `query` and was thinking that it might be good to have `query` as the public interface and `_query` as the private, where something like `_authenticate` would call `_query` and specify never to refresh - something that might be important if the ttls on sessions were tightened up in the future if we had a problem with too many open connections on the server